actions.go

```diff
@@ -113,6 +120,57 @@ func AppCreateAction(clientConfig spinnaker.ClientConfig) cli.ActionFunc {
 	}
 }
 
+// AppDeleteAction delete an application
+func AppDeleteAction(clientConfig spinnaker.ClientConfig) cli.ActionFunc {
+	return func(cc *cli.Context) error {
+		appName := cc.Args().Get(0)
+
+		config := make(map[string]interface{})
+
+		config["name"] = appName
+
+		deleteAppJob := spinnaker.ApplicationJob{
+			Application: config,
+			Type:        "deleteApplication",
+		}
+
+		deleteApp := spinnaker.Task{
+			Application: appName,
+			Description: "Delete Application: " + appName,
+			Job:         []interface{}{deleteAppJob},
+		}
+
+		client, err := clientFromContext(cc, clientConfig)
+		if err != nil {
+			return errors.Wrapf(err, "creating spinnaker client")
+		}
+
+		logrus.Info("Sending delete app task")
+		ref, err := client.ApplicationSubmitTask(appName, deleteApp)
+		if err != nil {
+			return errors.Wrapf(err, "submitting task")
+		}
+
+		resp, err := client.PollTaskStatus(ref.Ref, 1*time.Minute)
```

Comment:
I think since we're adding the ability to change the poll timeout duration for other tasks, we should do it across the board.